src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ExcavatorProfitTracker.kt

```diff
@@ -0,0 +1,182 @@
+import at.hannibal2.skyhanni.utils.CollectionUtils.addAsSingletonList
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getPrice
+import at.hannibal2.skyhanni.utils.NumberUtil
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.tracker.ItemTrackerData
+import at.hannibal2.skyhanni.utils.tracker.SkyHanniItemTracker
+import com.google.gson.annotations.Expose
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class ExcavatorProfitTracker {
+
+    private val config get() = SkyHanniMod.feature.mining.fossilExcavator.profitTracker
+
+    private val tracker = SkyHanniItemTracker(
+        "Fossil Excavation Profit Tracker",
+        { Data() },
+        { it.mining.fossilExcavatorProfitTracker }) { drawDisplay(it) }
+
+    class Data : ItemTrackerData() {
+        override fun resetItems() {
+            timesExcavated = 0
+        }
+
+        override fun getDescription(timesGained: Long): List<String> {
+            val percentage = timesGained.toDouble() / timesExcavated
+            val dropRate = LorenzUtils.formatPercentage(percentage.coerceAtMost(1.0))
+            return listOf(
+                "§7Dropped §e${timesGained.addSeparators()} §7times.",
+                "§7Your drop rate: §c$dropRate."
+            )
+        }
+
+        override fun getCoinName(item: TrackedItem) = "<no coins>"
+
+        override fun getCoinDescription(item: TrackedItem): List<String> {
+            return listOf(
+                "<no coins>"
+            )
+        }
+
+        @Expose
+        var timesExcavated = 0L
+        var glacitePowderGained = 0L
+        var fossilDustGained = 0L
+    }
+
+    private val scrapItem = "SUSPICIOUS_SCRAP".asInternalName()
+
+    private fun drawDisplay(data: Data): List<List<Any>> = buildList {
+        addAsSingletonList("§e§lFossil Excavation Profit Tracker")
+        var profit = tracker.drawItems(data, { true }, this)
+
+        val timesExcavated = data.timesExcavated
+        addAsSingletonList(
+            Renderable.hoverTips(
+                "§7Times excavated: §e${timesExcavated.addSeparators()}",
+                listOf("§7You excavated §e${timesExcavated.addSeparators()} §7times.")
+            )
+        )
+
+        // TODO use same price source as profit tracker
+        profit = addScrap(timesExcavated, profit)
+        if (config.trackGlacitePowder) {
+            addGlacitePowder(data)
+        }
+
+        addAsSingletonList(tracker.addTotalProfit(profit, data.timesExcavated, "excarvation"))
```

Comment:
should be "excavation" not "excarvation"
```suggestion
        addAsSingletonList(tracker.addTotalProfit(profit, data.timesExcavated, "excavation"))
```